### DISCUSSION TITLE
Server Side Pagination

### DISCUSSION THREAD

**Discussion by espeoceka** [zauberzeug/nicegui#707] [Q&A]
### Question

Hey is there any shortcut way to display very long lists with ui.table/ ui.aggrid ? ( maybe there is a example i missed )

I got quite simple list where first column is  userid and it should link to /user?id=userid.
First thing i noticed is that ui.table cant simply display html or ui.button. So first i used aggrid to:

```py
grid = ui.aggrid({
    'columnDefs': [
        {'headerName': 'Name', 'field': 'name','filter': 'agTextColumnFilter','floatingFilter': True,'sortable':True },
        {'headerName': 'Age', 'field': 'age'},
        {'headerName': 'Options', 'field': 'options'}
    ],
    'rowData': [
        {'name': 'Alice', 'age': 18,
         'options':'<div class="flex gap-3 items-center justify-center"><a class="underline text-blue-500" href="/user/contact_lists">Alice</a><a class="q-btn text-xs q-btn--standard q-btn--rectangle bg-primary text-white q-btn--actionable q-focusable q-hoverable">TEST</a></div>'         },
        {'name': 'Bob', 'age': 21},
        {'name': 'Carol', 'age': 42},
    ],
    'rowSelection': 'multiple',
},html_columns=[0,2],theme='material')
```
       
But now i really dont know how to grasp this.... should i use aggrid api ? or maybe theres better way.... ( im trying to avoid inventing wheel but its quite tricky to get there with just examples  )

**Comment by falkoschindler** [comment]
Hi @espeoceka, could you please clarify what your actual question is? AG Grids have been proven to be effective in handling large tables, as demonstrated in https://github.com/zauberzeug/nicegui/discussions/632. If you haven't tried it yet, I would recommend giving it a go and sharing any specific issues you might encounter, so we can better assist you.

**Reply by espeoceka** [reply]
Well i just noticed that all of server side features of aggrid are in enterprise version....So this topic can be safely closed

I got simply few places where i need to search on big set of data on mongodb... somehow one approach lack html support or elements embeddings to display proper buttons and aggrid lack of server side processing in community edition....
